Dockerfile.product

```diff
@@ -0,0 +1,46 @@
+FROM openshift/origin-base
+
+ENV GOPATH=/go
+
+RUN mkdir -p /go/src/github.com/openshift/console/
+ADD . /go/src/github.com/openshift/console/
+WORKDIR /go/src/github.com/openshift/console/
+ADD etc/ssl /etc/ssl
+
+# TODO: Switch to multistage builds as soon as we can.
+#
+# Multi stage builds are not yet supported for product builds. For now, install
+# dependencies and do our best to clean up after building.
+#
+# Node.js 8+ is required, but EPEL only has Node.js 6.
+#
+# `rm -rf /tmp/*` is needed to remove chromedriver and other temporary files
+# created while building frontend assets.
+
+RUN curl --silent --location https://rpm.nodesource.com/setup_8.x | bash - && \
+    curl --silent --location https://dl.yarnpkg.com/rpm/yarn.repo | tee /etc/yum.repos.d/yarn.repo && \
+    INSTALL_PKGS="git golang nodejs yarn" && \
```

Comment:
From what I understand we're going to start using go-toolset-7 to get go 1.10 (apparently golang is EOL at 1.9). This is an SCL so you'll need to enable the SCL in order to build. I don't actually know anything else about it yet.

Reply:
I wonder if we don't have nodejs 8 and yarn in an SCL somewhere too. That would be preferable to building from source. Even though we're going to end up sucking down a lot of JS code.

Reply:
Working on a scheme to ease the go-toolset-7 transition. As for the other tools, I'd say we should look for internal builds ASAP, but it need not hold up this PR.